cfme/infrastructure/virtual_machines.py

```diff
@@ -86,46 +86,110 @@
-vms_tree = partial(accordion.tree, "VMs")
-templates_tree = partial(accordion.tree, "Templates")
-
-
-def reset_page():
-    tb.select("Grid View")
-    from cfme.web_ui import paginator
-    if sel.is_displayed(search_box.search_field):
-        search.ensure_normal_search_empty()
-    if paginator.page_controls_exist():
-        # paginator.results_per_page(1000)
-        paginator.check_all()
-        paginator.uncheck_all()
 
 
 drift_table = CheckboxTable("//th[normalize-space(.)='Timestamp']/ancestor::table[1]")
 
 
-class InfraVmDetailsToolbar(View):
-    configuration = Dropdown("Configuration")
-    policy = Dropdown("Policy")
-    lifecycle = Dropdown("Lifecycle")
+class InfraVmDetailsToolbar(VMToolbar):
     monitoring = Dropdown("Monitoring")
-    power = Dropdown("Power")
     access = Dropdown("Access")
+    reload = Button(title='Reload current display')
+
+
+class VmsTemplatesAccordion(View):
+    """
+    The accordion on the Virtual Machines page
+    """
+    @View.nested
+    class vmstemplates(Accordion):  # noqa
+        ACCORDION_NAME = 'VMs & Templates'
+        tree = ManageIQTree()
+
+    @View.nested
+    class vms(Accordion):  # noqa
+        ACCORDION_NAME = 'VMs'
+        tree = ManageIQTree()
+
+    @View.nested
+    class templates(Accordion):  # noqa
+        ACCORDION_NAME = 'Templates'
+        tree = ManageIQTree()
+
 
+class InfraVMView(BaseLoggedInPage):
+    """Base view for header/nav check, inherit for navigatable views"""
 
-class InfraVmDetailsView(BaseLoggedInPage):
-    # TODO this is only minimal implementation for toolbar access through widgetastic
+    including_entities = View.include(VMEntities, use_parent=True)
+
+    @View.nested
+    class vmstemplates(Accordion):  # noqa
+        ACCORDION_NAME = 'VMs & Templates'
+        tree = ManageIQTree()
+
+    @View.nested
+    class vms(Accordion):  # noqa
+        ACCORDION_NAME = 'VMs'
+        tree = ManageIQTree()
+
+    @View.nested
+    class templates(Accordion):  # noqa
+        ACCORDION_NAME = 'Templates'
+        tree = ManageIQTree()
+
+    @property
+    def in_infra_vms(self):
+        return (self.logged_in_as_current_user and
+                self.navigation.currently_selected == ['Compute', 'Infrastructure',
+                                                       'Virtual Machines'] and
+                match_location(controller='vm_infra', title='Virtual Machines'))
+
+
+class VmsTemplatesAllView(InfraVMView):
+    """
+    The collection page for instances
+    """
+    @property
+    def is_displayed(self):
+        return self.in_infra_vms
+
+    toolbar = View.nested(VMToolbar)
+    sidebar = View.nested(VmsTemplatesAccordion)
+    including_entities = View.include(VMEntities, use_parent=True)
+
+
+class InfraVmDetailsView(InfraVMView):
     toolbar = View.nested(InfraVmDetailsToolbar)
+    sidebar = View.nested(VmsTemplatesAccordion)
+    entities = View.nested(VMDetailsEntities)
```

Comment:
It looks like some infra VM's will have additional `SummaryTable` widgets in their details view.

I'm seeing 'Datastore Allocation Summary' and 'Datastore Actual Usage Summary' when looking at vsphere VM's.